TAScheduler/UnitTests/tests.py

```diff
@@ -178,50 +189,827 @@ class TestCourseGetCrseInfo(TestCase):  # Randall
+            num_of_sections=3,
+            modality="online",
+            credits=3
+        )
+        tmpcourse.save()
+
+        tmpsection = Section.objects.create(
+            section_id=1011,
+            course=tmpcourse,
+            location="Cool place",
+            meeting_time=datetime.datetime
+        )
+        tmpsection.save()
+        self.section = SectionObj(tmpsection)
+
+    def test_get_id(self):
+        self.assertEquals(self.section.getID(), 1011, "getID() did not retrieve correct section_id")
+
+    # Maybe test for section_id = None, but I don't think section can be made without
 
 
 class TestSectionGetParentCourse(TestCase):  # Joe
-    pass
+    section = None
+    course = None
+
+    # noinspection DuplicatedCode
+    def setUp(self):
+        self.course = Course.objects.create(
+            course_id=100,
+            semester="fall 2023",
+            name="testCourse",
+            description="test",
+            num_of_sections=3,
+            modality="online",
+            credits=3
+        )
+        self.course.save()
+
+        tmpsection = Section.objects.create(
+            section_id=1011,
+            course=self.course,
+            location="Cool place",
+            meeting_time=datetime.datetime
+        )
+        tmpsection.save()
+        self.section = SectionObj(tmpsection)
+
+    def test_get_parent_course(self):
+        self.assertEquals(self.section.getParentCourse(), self.course,
+                          "getParentCourse() did not retrieve correct course")
 
 
 class TestLabInit(TestCase):
-    pass
+    lab = None
+    info = None
+
+    # noinspection DuplicatedCode
+    def setUp(self):
+        tempuser = User(
+            email_address="test@test.com",
+            password="password",
+            first_name="first",
+            last_name="last",
+            home_address="Your mom's house",
+            phone_number=1234567890
+        )
+        tempuser.save()
+
+        tmpta = TA.objects.create(
+            user=tempuser,
+            grader_status=True
+        )
+        tmpta.save()
+
+        tmpcourse = Course.objects.create(
+            course_id=100,
+            semester="fall 2023",
+            name="testCourse",
+            description="test",
+            num_of_sections=3,
+            modality="online",
+            credits=3
+        )
+        tmpcourse.save()
+
+        tmpsection = Section.objects.create(
+            section_id=1011,
+            course=tmpcourse,
+            location="Cool place",
+            meeting_time=datetime.datetime
+        )
+        tmpsection.save()
+
+        tmplab = Lab.objects.create(
+            section=tmpsection,
+            ta=tmpta
+        )
+
+        self.info = {
+            "ta", tmpta,
+            "section", tmpsection
+        }
+
+        self.lab = LabObj(tmplab)
+
+    def test_lab_make(self):
+        self.assertIsNotNone(self.lab.__init__(self.info), "__init__ failed in making Lab")
+
```

Comment:
Also probably should add some tests for when __init__ so provided bad input ie not a Lab object from the database